celery/backends/dynamodb.py

```diff
@@ -206,6 +226,161 @@ def _get_or_create_table(self):
             else:
                 raise e
 
+    def _has_ttl(self):
+        """Return the desired Time to Live config; True means enabled."""
+        return self.time_to_live_seconds is not None
+
+    def _get_ttl_specification(self, ttl_attr_name):
+        """Get the boto3 structure describing the DynamoDB TTL specification."""
+        return {
+            'TableName': self.table_name,
+            'TimeToLiveSpecification': {
+                'Enabled': self._has_ttl(),
+                'AttributeName': ttl_attr_name
+            }
+        }
+
+    def _get_table_ttl_description(self):
+        # Get the current TTL description.
+        try:
+            description = self._client.describe_time_to_live(
+                TableName=self.table_name
+            )
+            status = description['TimeToLiveDescription']['TimeToLiveStatus']
+        except ClientError as e:
+            error_code = e.response['Error'].get('Code', 'Unknown')
+            error_message = e.response['Error'].get('Message', 'Unknown')
+            logger.error((
+                'Error describing Time to Live on DynamoDB table {table}: '
+                '{code}: {message}'
+            ).format(
+                table=self.table_name,
+                code=error_code,
+                message=error_message,
+            ))
+            raise e
+
+        return description
+
+    def _set_table_ttl(self):
+        """Enable or disable Time to Live on the table."""
+
+        # Verify client support for the DynamoDB TTL methods.
+        for method in ('update_time_to_live', 'describe_time_to_live'):
+            if not hasattr(self._client, method):
+                message = (
+                    "boto3 method '{method}' not found; ensure that "
+                    "boto3>=1.9.178 and botocore>=1.12.178 are installed"
+                ).format(method=method)
+                if not self._has_ttl():
```

Comment:
Can't this part stand on its own as a guard? Even at `debug` level, the log can be a distraction when its not really used.

Reply:
+  I would move the boto validation code elsewhere

And then you could do:

    self.validate_boto_version()

It would make the code more direct. This method verification is a necessary nuisance which isn't at the heart of this operation and has enough code to justify its own method.

(it would also be nice to see all missing methods and not just the first, but that is also not a big deal).

Reply:
Addressed.